sympy/functions/combinatorial/factorials.py

```diff
@@ -1100,3 +1110,107 @@ def _eval_is_nonnegative(self):
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         from sympy.functions.special.gamma_functions import gamma
         return self.rewrite(gamma)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+
+
+def _binomial_modulo_p_pow(n, m, p, q):
```

Comment:
Maybe it would make more sense to put this helper in sympy.ntheory somewhere.

Reply:
Some of the helper functions could be independently useful there too perhaps.

Reply:
Minor question: should `binomial_mod` be exposed in `sympy.__init__.py` so that you can `from sympy import binomial_mod`?

Reply:
For now let's leave it out. You can access this functionality using Mod on a Binomial (it's unfortunate that you have to evaluate=False to prevent it from trying to evaluate big numbers, but that's a separate issue, c.f. https://github.com/sympy/sympy/issues/6835).